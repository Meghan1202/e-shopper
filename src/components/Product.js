import React from 'react';
import './Product.css';
import PropTypes from 'prop-types';
import Quantity from './Quantity';

const Product = ({
  productName, productPrice, productId, productQuantity,
}) => (
  <div className="card">
    <img id="banana__img" src="https://upload.wikimedia.org/wikipedia/commons/8/8a/Banana-Single.jpg" alt="Banana" />
    <div className="container">
      <h4>
        { productId}
        <b>{productName}</b>
      </h4>
      <p>
        One kg for
        {productPrice}
      </p>
      <div className="inner__card">
        <p>
          Quantity:
          { productQuantity}
        </p>
        <Quantity count={20} />
      </div>
    </div>
  </div>
);

Product.propTypes = {
  productPrice: PropTypes.string.isRequired,
  productId: PropTypes.string.isRequired,
  productName: PropTypes.string.isRequired,
  productQuantity: PropTypes.string.isRequired,
};

export default Product;
