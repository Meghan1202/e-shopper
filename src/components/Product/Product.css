.card {
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  transition: 0.3s;
  border-radius: 1rem;
  max-width: fit-content;
  padding: 1rem;
  margin: 2rem;
}

.card:hover {
  box-shadow: 0 8px 16px 0 rgba(233, 200, 55, 0.308);
}
.container {
  padding: 2px;
}

.banana__img{
    width: 10rem;
    height: 10rem;
    border-radius: 0.5rem;
}

.inner__card{
  display: flex;
  align-items: center;
  flex-direction: column;
  background-color: lightgrey;
  border-radius: 16px;
}