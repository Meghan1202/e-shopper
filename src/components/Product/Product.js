import React from 'react';
import './Product.css';
import PropTypes from 'prop-types';
import Quantity from '../Quantity/Quantity';
import SoldOut from '../SoldOut/SoldOut';

const Product = ({
  productName, productPrice, productQuantity, onIncrement, onDecrement, count, companyName, imgSrc,
}) => (
  <div data-testid="product-card" className="card">
    <img data-testid="product-image" id="banana__img" src={imgSrc} alt="Banana" />
    <div className="container">
      <h4>
        {companyName}
      </h4>
      <p><b>{productName}</b></p>
      <p>
        In Stock:
        { productQuantity}
      </p>
      <div className="inner__card">
        <p>
          {productPrice}
          /-
        </p>
        {productQuantity > 0
          ? <Quantity count={count} onIncrement={onIncrement} onDecrement={onDecrement} />
          : <SoldOut />}
      </div>
    </div>
  </div>
);

Product.propTypes = {
  productPrice: PropTypes.number.isRequired,
  productName: PropTypes.string.isRequired,
  productQuantity: PropTypes.number.isRequired,
  imgSrc: PropTypes.string.isRequired,
  count: PropTypes.number.isRequired,
  companyName: PropTypes.string.isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};

export default Product;
