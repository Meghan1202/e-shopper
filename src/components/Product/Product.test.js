import {
  fireEvent, render, screen, waitFor,
} from '@testing-library/react';
import React from 'react';
import Product from './Product';

describe(Product.name, () => {
  test('should match snapshot', () => {
    const { container } = render(<Product productName="Banana" productPrice={40} productQuantity="2" onIncrement={jest.fn()} onDecrement={jest.fn()} count={1} companyName="Fresho" imgSrc="abc" />);
    expect(container).toMatchSnapshot();
  });
  test;
});
