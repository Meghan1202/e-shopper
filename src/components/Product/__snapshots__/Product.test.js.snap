// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Banana"
      id="banana__img"
      src="abc"
    />
    <div
      class="container"
    >
      <h4>
        Fresho
      </h4>
      <p>
        <b>
          Banana
        </b>
      </p>
      <p>
        Quantity:
        2
      </p>
      <div
        class="inner__card"
      >
        <p>
          40
          /-
        </p>
        <div
          class="quantity__count"
          data-testid="quantity-count"
        >
          <button
            type="button"
          >
            -
          </button>
          <p>
            1 in Basket
          </p>
          <button
            type="button"
          >
             + 
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
